testpb/1.pulsar.go

```diff
@@ -976,7 +1007,1122 @@ func (x *fastReflection_A) IsValid() bool {
+				} else if wireType == 2 {
+					var packedLen int
+					for shift := uint(0); ; shift += 7 {
+						if shift >= 64 {
+							return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrIntOverflow
+						}
+						if iNdEx >= l {
+							return protoiface.UnmarshalOutput{Flags: input.Flags}, io.ErrUnexpectedEOF
+						}
+						b := dAtA[iNdEx]
+						iNdEx++
+						packedLen |= int(b&0x7F) << shift
+						if b < 0x80 {
+							break
+						}
+					}
+					if packedLen < 0 {
+						return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrInvalidLength
+					}
+					postIndex := iNdEx + packedLen
+					if postIndex < 0 {
+						return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrInvalidLength
+					}
+					if postIndex > l {
+						return protoiface.UnmarshalOutput{Flags: input.Flags}, io.ErrUnexpectedEOF
+					}
+					var elementCount int
+					if elementCount != 0 && len(x.LIST_ENUM) == 0 {
+						x.LIST_ENUM = make([]Enumeration, 0, elementCount)
+					}
+					for iNdEx < postIndex {
+						var v Enumeration
+						for shift := uint(0); ; shift += 7 {
+							if shift >= 64 {
+								return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrIntOverflow
+							}
+							if iNdEx >= l {
+								return protoiface.UnmarshalOutput{Flags: input.Flags}, io.ErrUnexpectedEOF
+							}
+							b := dAtA[iNdEx]
+							iNdEx++
+							v |= Enumeration(b&0x7F) << shift
+							if b < 0x80 {
+								break
+							}
+						}
+						x.LIST_ENUM = append(x.LIST_ENUM, v)
+					}
+				} else {
+					return protoiface.UnmarshalOutput{Flags: input.Flags}, fmt.Errorf("proto: wrong wireType = %d for field LIST_ENUM", wireType)
+				}
+			case 23:
+				if wireType != 2 {
+					return protoiface.UnmarshalOutput{Flags: input.Flags}, fmt.Errorf("proto: wrong wireType = %d for field Imported", wireType)
+				}
+				var msglen int
+				for shift := uint(0); ; shift += 7 {
+					if shift >= 64 {
+						return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrIntOverflow
+					}
+					if iNdEx >= l {
+						return protoiface.UnmarshalOutput{Flags: input.Flags}, io.ErrUnexpectedEOF
+					}
+					b := dAtA[iNdEx]
+					iNdEx++
+					msglen |= int(b&0x7F) << shift
+					if b < 0x80 {
+						break
+					}
+				}
+				if msglen < 0 {
+					return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrInvalidLength
+				}
+				postIndex := iNdEx + msglen
+				if postIndex < 0 {
+					return protoiface.UnmarshalOutput{Flags: input.Flags}, runtime.ErrInvalidLength
+				}
+				if postIndex > l {
+					return protoiface.UnmarshalOutput{Flags: input.Flags}, io.ErrUnexpectedEOF
+				}
+				if x.Imported == nil {
+					x.Imported = &descriptorpb.FileDescriptorProto{}
+				}
+				if unmarshal, ok := interface{}(x.Imported).(interface {
+					Unmarshal([]byte) error
+				}); ok {
+					if err := unmarshal.Unmarshal(dAtA[iNdEx:postIndex]); err != nil {
+						return protoiface.UnmarshalOutput{Flags: input.Flags}, err
```

Comment:
we should not call unmarshal directly here, we should use unmarshal options always.

Reply:
oops, this code shouldn't even be there actually, it was leftover from when it was casting OneOfs to the unmarshal interface. removed this block from codegen